docs/programming_guide/fed_job_api.rst

```diff
@@ -0,0 +1,244 @@
+* A :class:`ValidationJsonGenerator<nvflare.app_common.widgets.validation_json_generator.ValidationJsonGenerator>` is automatically added for creating json validation results.
+* If PyTorch and TensorBoard are supported, then :class:`TBAnalyticsReceiver<nvflare.app_common.pt.tb_receiver.TBAnalyticsReceiver>` is automatically added to receives analytics data to save to TensorBoard. Other types of receivers can be added as components with :func:`to<nvflare.job_config.fed_job.FedJob.to>`.
+
+Example:
+
+.. code-block:: python
+
+  controller = FedAvg(
+      num_clients=n_clients,
+      num_rounds=num_rounds,
+  )
+  job.to(controller, "server")
+
+
+Executor
+--------
+
+If the object is an :class:`Executor<nvflare.apis.executor.Executor>`, the executor is added to the client app executors.
+
+* The ``tasks`` parameter specifies the tasks that the executor is defined the handle.
+* The ``gpu`` parameter specifies which gpus to use for simulating the run on the target.
+* If the object is a :class:`ScriptExecutor<nvflare.app_common.executors.script_executor.ScriptExecutor>`, the task_script_path will be added to the external scripts to be included in the custom directory.
+* The :class:`ConvertToFedEvent<nvflare.app_common.widgets.convert_to_fed_event.ConvertToFedEvent>` widget is automatically added to convert local events to federated events.
+
+Example:
+
+.. code-block:: python
+
+  executor = ScriptExecutor(task_script_path="src/cifar10_fl.py", task_script_args="")
+  job.to(executor, "site-1", tasks=["train"], gpu=0)
+
+
+Script (str)
+------------
+
+If the object is a str, it is treated as an external script and will be included in the custom directory.
+
+Example:
+
+.. code-block:: python
+
+  job.to("src/cifar10_fl.py", "site-1")
+
+
+Filter
+------
+
+If the object is a :class:`Filter<nvflare.apis.filter.Filter>`, users must specify the ``filter_type``
+as either FilterType.TASK_RESULT (flow from executor to controller) or FilterType.TASK_DATA (flow from controller to executor).
+
+The filter will be added task_data_filters and task_result_filters accordingly and be applied to the specified ``tasks``.
+
+Example:
+
+.. code-block:: python
+
+  pp_filter = PercentilePrivacy(percentile=10, gamma=0.01)
+  job.to(pp_filter, "site-1", tasks=["train"], filter_type=FilterType.TASK_RESULT)
+
+
+Model
+-----
+If the object is a common model type, a corresponding persistor will automatically be configured with the model.
+
+For PyTorch models (``torch.nn.Module``) we add a :class:`PTFileModelPersistor<nvflare.app_opt.pt.file_model_persistor.PTFileModelPersistor>` and
+:class:`PTFileModelLocator<nvflare.app_opt.pt.file_model_locator.PTFileModelLocator>`, and for TensorFlow models (``tf.keras.Model``) we add a :class:`TFModelPersistor<nvflare.app_opt.tf.model_persistor.TFModelPersistor>`.
+
+Example:
+
+.. code-block:: python
+
+  job.to(Net(), "server")
+
+For unsupported models, the model and persistor can be added as components.
+
+
+Components
```

Comment:
do we need to talk about client side controller either here or in Controller section?

Reply:
good idea I will add a section about this